### PR TITLE
add app label to pods

### PR DESCRIPTION
**Pull request by bewebi**
# Description

- Add `app: gloo` label to deployments' pod templates and to deployments' `matchLabels` selector
  - Update tests accordingly
- Add env var to cloudbuild to indicate to helm tests whether the suite is being run in CI where slightly modified logic needs to be applied  

# Context

We want to allow for (and make default) a Prometheus scraping config that is limited to Gloo Edge pods. In order to do so it is helpful to label all Edge pods with `app: gloo`

In the process of writing this PR I came across #7063 as well and have included a fix for it

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/7063

### CODE REVIEW

**Review by kdorosh — COMMENTED** [review]
reasonable to me

**Comment by jenshu** [comment]
there are a couple other deployments that don't have `app: gloo`, do those need to be fixed too?
(i searched `kind: Deployment` from `install/helm`)

File: install/test/helm_test.go [hunk 1]
```diff
@@ -2575,7 +2576,7 @@ spec:
 							daemonSet.Spec.Template.Spec.HostNetwork = true
 						})
 
-						It("creates a daemonset", func() {
+						FIt("creates a daemonset", func() {
```
Comment (jenshu) [inline install/test/helm_test.go:2579]: unfocus

**Comment by bewebi** [comment]
solo-io/solo-projects#4046 introduces a scrape config that does not require the `app` label to be set
Therefore this PR is not necessary

**Pull request closed without merging**